### PR TITLE
Friend: Stub GetPlayHistoryRegistrationKey

### PR DESCRIPTION
**Pull request by german77**
Stubs GetPlayHistoryRegistrationKey used by BLAZBLUE CROSS TAG BATTLE

### CODE REVIEW

File: src/core/hle/service/friend/friend.cpp [hunk 1]
```diff
@@ -153,6 +153,19 @@ class IFriendService final : public ServiceFramework<IFriendService> {
         rb.Push(RESULT_SUCCESS);
     }
 
+    void GetPlayHistoryRegistrationKey(Kernel::HLERequestContext& ctx) {
+        // Stub used by BLAZBLUE CROSS TAG BATTLE
+        IPC::RequestParser rp{ctx};
+        const auto bool_value = rp.Pop<bool>();
+        const auto uuid = rp.PopRaw<Common::UUID>();
+
+        LOG_WARNING(Service_ACC, "(STUBBED) called bool={} uuid={}", bool_value, uuid.Format());
```
Comment (Morph1984) [inline src/core/hle/service/friend/friend.cpp:162]:
```suggestion
        const auto unknown_bool = rp.Pop<bool>();
        const auto uuid = rp.PopRaw<Common::UUID>();

        LOG_WARNING(Service_ACC, "(STUBBED) called, unknown_bool={}, uuid={}", unknown_bool, uuid.Format());
```

File: src/core/hle/service/friend/friend.cpp [hunk 2]
```diff
@@ -153,6 +153,19 @@ class IFriendService final : public ServiceFramework<IFriendService> {
         rb.Push(RESULT_SUCCESS);
     }
 
+    void GetPlayHistoryRegistrationKey(Kernel::HLERequestContext& ctx) {
+        // Stub used by BLAZBLUE CROSS TAG BATTLE
```
Comment (ogniK5377) [inline src/core/hle/service/friend/friend.cpp:157]: This can be omitted

File: src/core/hle/service/friend/friend.cpp [hunk 3]
```diff
@@ -153,6 +153,19 @@ class IFriendService final : public ServiceFramework<IFriendService> {
...
+        IPC::RequestParser rp{ctx};
+        const auto bool_value = rp.Pop<bool>();
```
Comment (ogniK5377) [inline src/core/hle/service/friend/friend.cpp:159]:
```suggestion
        const auto local_play = rp.Pop<bool>();
```

File: src/core/hle/service/friend/friend.cpp (hunk 1 above)
Comment (ogniK5377) [inline src/core/hle/service/friend/friend.cpp:162]:
```suggestion
        LOG_WARNING(Service_Friend, "(STUBBED) called local_play={} uuid={}", local_play, uuid.Format());
```

**Review by ogniK5377 — CHANGES_REQUESTED** [review]
Also change all the log classes in the file to `Service_Friend`, Thanks!

File: src/core/hle/service/friend/friend.cpp [hunk 4]
```diff
@@ -153,6 +153,19 @@ class IFriendService final : public ServiceFramework<IFriendService> {
...
+        const auto uuid = rp.PopRaw<Common::UUID>();
+
+        LOG_WARNING(Service_ACC, "(STUBBED) called bool={} uuid={}", bool_value, uuid.Format());
+
+        IPC::ResponseBuilder rb{ctx, 3};
+        rb.Push(RESULT_SUCCESS);
+        rb.Push<u32>(0);
```
Comment (ogniK5377) [inline src/core/hle/service/friend/friend.cpp:166]:
```suggestion
        IPC::ResponseBuilder rb{ctx, 2};
        rb.Push(RESULT_SUCCESS);
```

**Review by ogniK5377 — CHANGES_REQUESTED** [review]
Also, it looks as if only a buffer is written out, not any inline values and not a u32.

File: src/core/hle/service/friend/friend.cpp (hunk 1 above)
Comment (german77) [reply src/core/hle/service/friend/friend.cpp:162]: Most logs here are Service_ACC Should I change them for Friend?

**Review by bunnei — APPROVED** [review]

**Pull request merged**